src/main.rs

```diff
@@ -137,6 +159,34 @@ async fn main() -> Result<()> {
     }
 }
 
+async fn tls_config_from_params(
+    private_key: Option<PathBuf>,
+    certificate: Option<PathBuf>,
+    websocket: bool,
+) -> Result<Option<tls::Config>> {
+    let tls_config = match (private_key, certificate) {
+        (Some(pk), Some(cert)) => {
+            if !websocket {
+                tracing::warn!("The provided SSL parameters won't have any affect, because you did not activate websockets");
+            }
+
+            let pk = fs::read(pk).await?;
+            let cert = fs::read(cert).await?;
+
+            let pk = PrivateKey::new(pk);
+            let cert = Certificate::new(cert);
+
+            let tls_config = tls::Config::new(pk, vec![cert])?;
+
+            Some(tls_config)
+        }
+        (None, None) => None,
+        _ => bail!("Server private key and certificate both have to be provided"),
+    };
+
+    Ok(tls_config)
```

Comment:
Probably a bit easier to understand to use defensive programming here and exit early for all the non-happy-path cases:

```suggestion
    let (pk, cert) = match (private_key, certificate) {
	    (None, None) => return None,
        (Some(pk), Some(cert)) => (pk, cert),
        _ => bail!("Server private key and certificate both have to be provided"),
    };

    if !websocket {
        tracing::warn!("The provided SSL parameters won't have any affect, because you did not activate websockets");
        return None; // I added this, wasn't there in the previous version
    }

    let pk = fs::read(pk).await?;
    let cert = fs::read(cert).await?;

    let pk = PrivateKey::new(pk);
    let cert = Certificate::new(cert);

    let tls_config = tls::Config::new(pk, vec![cert])?;

    Ok(tls_config)
```

Reply:
The websocket check could in theory be outside, it is orthogonal to creating a new tls-config.

Reply:
yes and now - at the moment the TLC config is only used in case websockets are specified so somehow I find it nicer to keep it in the function so main stays cleaner. I'd keep it for now, we can change it later if we have a strong feeling about it.

Reply:
note: the code changes proposed above were applied :)